python/tests/test_highlevel.py

```diff
@@ -2719,34 +2743,76 @@ def convert(v):
             assert str(individual.flags) == splits[1]
             assert ",".join(map(str, individual.location)) == splits[2]
             assert ",".join(map(str, individual.parents)) == splits[3]
-            assert tests.base64_encode(individual.metadata) == splits[4]
+            if isinstance(individual.metadata, bytes) and base64_metadata:
+                assert tests.base64_encode(individual.metadata) == splits[4]
+            else:
+                assert repr(individual.metadata) == splits[4]
+
+    def verify_populations_format(
+        self, ts, populations_file, precision, base64_metadata
+    ):
+        """
+        Verifies that the populations we output have the correct form.
+        """
+
+        def convert(v):
+            return "{:.{}f}".format(v, precision)
+
+        output_populations = populations_file.read().splitlines()
+        print(output_populations)
+        assert len(output_populations) - 1 == ts.num_populations
+        assert list(output_populations[0].split()) == [
+            "id",
+            "metadata",
+        ]
+        for population, line in zip(ts.populations(), output_populations[1:]):
+            splits = line.split("\t")
+            assert str(population.id) == splits[0]
+            if isinstance(population.metadata, bytes) and base64_metadata:
+                assert tests.base64_encode(population.metadata) == splits[1]
+            else:
+                assert repr(population.metadata) == splits[1]
 
     def test_output_format(self):
         for ts in get_example_tree_sequences():
             for precision in [2, 7]:
-                nodes_file = io.StringIO()
-                edges_file = io.StringIO()
-                sites_file = io.StringIO()
-                mutations_file = io.StringIO()
-                individuals_file = io.StringIO()
-                ts.dump_text(
-                    nodes=nodes_file,
-                    edges=edges_file,
-                    sites=sites_file,
-                    mutations=mutations_file,
-                    individuals=individuals_file,
-                    precision=precision,
-                )
-                nodes_file.seek(0)
-                edges_file.seek(0)
-                sites_file.seek(0)
-                mutations_file.seek(0)
-                individuals_file.seek(0)
-                self.verify_nodes_format(ts, nodes_file, precision)
-                self.verify_edges_format(ts, edges_file, precision)
-                self.verify_sites_format(ts, sites_file, precision)
-                self.verify_mutations_format(ts, mutations_file, precision)
-                self.verify_individuals_format(ts, individuals_file, precision)
+                for base64_metadata in [True, False]:
```

Comment:
I don't think there's any point in taking the cross product of precision and base64_metadata, it's not really testing anything useful. I think we're going to need to move away from the old "sledgehammer" approach to testing, and be more selective about how we do things (and more obvious, using examples).

Feels like a little bit of light refactoring here and pytest.mark.parametrize would help? I've been trying to modernize test methods that I encounter as I go, otherwise it'll never get done.